### PR TITLE
Documentation webapi

### PR DESCRIPTION
**Pull request by apenzk**
# Description of change

Added documentation for WebAPI

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Documentation 

## Change checklist

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: docs/tooling/webAPI_clientLib.md [hunk 1]
```diff
@@ -1 +1,69 @@
...
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, Response{Error: err.Error()})
+	}
+	return c.JSON(http.StatusOK, Response{ID: msg.ID().String()})
+}
+```
+As an example the JSON body   
+```
+{
+	"data":"HelloWor"
```
Comment (capossele) [inline docs/tooling/webAPI_clientLib.md:64]:
```suggestion
	"data":"HelloIOTA"
```

File: docs/tooling/webAPI_clientLib.md [hunk 2]
```diff
@@ -1 +1,69 @@
...
+}
+```
+can be sent to `http://127.0.0.1:8080/data`, which will issue a data message containing "HelloWor" (note that in this  example the data input is size limited.)
```
Comment (capossele) [inline docs/tooling/webAPI_clientLib.md:67]:
```suggestion
can be sent to `http://127.0.0.1:8080/data`, which will issue a data message containing "HelloIOTA" (note that in this example the data input is size limited.)
```

File: docs/tooling/webAPI_clientLib.md [hunk 3]
```diff
@@ -1 +1,69 @@
...
+```
+We can then use the Context to send a JSON response to the node: 
+```
+JSON(statuscode int, i interface{}) error
+```
+An implementation example is shown later for the POST method.
+
+## GET and POST 
+
+Two methods are currently employed. First, with `GET` we register a new GET route for a handler function. The handler is accessed via the address `path`. The handler for a GET method can set the node to perform certain actions.
```
Comment (capossele) [inline docs/tooling/webAPI_clientLib.md:28]:
```suggestion
Two methods are currently used. First, with `GET` we register a new GET route for a handler function. The handler is accessed via the address `path`. The handler for a GET method can set the node to perform certain actions.
```

**Review by capossele — APPROVED** [review]

File: docs/tooling/webAPI_clientLib.md (hunk 1 above)
Comment (apenzk) [reply docs/tooling/webAPI_clientLib.md:64]: This does not work. The string cannot currently be chosen arbitrarily: 
{
  "error": "code=400, message=illegal base64 data at input byte 8"
}

Comment (capossele) [reply docs/tooling/webAPI_clientLib.md:64]: Yeah I forgot that data is supposed to be base64..  Nevermind then

**Pull request merged**